extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,241 @@
+    prompt.textContent = promptInput.textContent;
+    prompt.classList.add('i-amphtml-story-quiz-prompt');
+    promptInput.remove();
+
+    const options = toArray(this.element.querySelectorAll('option'));
+    if (options.length < 2 || options.length > 4) {
+      dev().error(TAG, 'Improper number of options');
+    }
+
+    this.quizEl_
+      .querySelector('.i-amphtml-story-quiz-prompt-container')
+      .appendChild(prompt);
+
+    options.forEach((option, index) => this.configureOption_(option, index));
+
+    if (this.element.children.length !== 0) {
+      dev().error(TAG, 'Too many children');
+    }
+  }
+
+  /**
+   * Creates an option container with option content,
+   * adds styling and answer choices,
+   * and adds it to the quiz element.
+   *
+   * @param {HTMLOptionElement} option
+   * @param {number} index
+   * @private
+   */
+  configureOption_(option, index) {
+    const convertedOption = buildOptionTemplate(option);
+
+    // Fill in the answer choice
+    convertedOption.querySelector(
+      '.i-amphtml-story-quiz-answer-choice'
+    ).textContent = answerChoiceOptions[index];
+
+    // Transfer the option information into a span then remove the option
+    const optionText = document.createElement('span');
+    optionText.classList.add('i-amphtml-story-quiz-option-text');
+    optionText.textContent = option.textContent;
+    convertedOption.append(optionText);
+
+    if (option.hasAttribute('correct')) {
+      convertedOption.setAttribute('correct', 'correct');
+    }
+    option.remove();
+
+    // Add the option to the quiz element
+    this.quizEl_
+      .querySelector('.i-amphtml-story-quiz-option-container')
+      .appendChild(convertedOption);
+  }
+
+  /**
+   * Attaches functions to each option to handle state transition.
+   *
+   * @private
+   */
+  initializeListeners_() {
+    // Add a listener for changes in the RTL state
+    this.storeService_.subscribe(
+      StateProperty.RTL_STATE,
+      rtlState => {
+        this.onRtlStateUpdate_(rtlState);
+      },
+      true /** callToInitialize */
+    );
+
+    // Add a click listener to the element to trigger the class change via tapping the prompt
+    this.quizEl_.addEventListener('click', e => this.handleTap_(e));
+  }
+
+  /**
+   * Handles a tap event on the quiz element
+   *
+   * @param {Event} e
+   * @private
+   */
+  handleTap_(e) {
+    if (this.hasReceivedResponse_) {
+      return;
+    }
+
+    const optionEl = closest(
+      e.target,
+      element => {
+        return element.classList.contains('i-amphtml-story-quiz-option');
+      },
+      this.quizEl_
+    );
+
+    if (optionEl) {
+      this.handleOptionSelection_(optionEl);
+    }
+  }
+
+  /**
+   * Triggers changes to quiz state on response interaction
+   *
+   * @param {!Element} optionEl
+   * @private
+   */
+  handleOptionSelection_(optionEl) {
+    optionEl.classList.add('i-amphtml-story-quiz-option-selected');
```

Comment:
Please wrap this into a `this.mutateElement(() => {...});` (all DOM mutations outside of `build/layoutCallback` have to)

Reply:
Ah right thanks for the reminder, done!